File: Python Vs Java BST Comparison/BST.py
import time
from random import randint

startTime = (time.time()) * 1000


class BSTNode:
    def __init__(self, num):
        self.left = None
        self.right = None
        self.key = num


def insert(root, bstNode):
    if root is None:
        root = bstNode

    else:
        if root.key < bstNode.key:
            if root.right is None:
                root.right = bstNode

            else:
                insert(root.right, bstNode)

        else:
            if root.left is None:
                root.left = bstNode

            else:
                insert(root.left, bstNode)


def displayInOrder(root):
    if root:
        displayInOrder(root.left)
        print(root.key)
        displayInOrder(root.right)


root = BSTNode(10)
i = 0

while i < 9:
    number = randint(0, 1000)
    insert(root, BSTNode(number))
    i += 1

displayInOrder(root)

endTime = (time.time()) * 1000
deltaTime = endTime - startTime
print(deltaTime)

File: Python Vs Java BST Comparison/BST.java
import java.util.Random;

public class BST{
    BSTNode root;

    public BST(){
        this.root = null;
    }

    void insert(int key){
        this.root =  insertRecur(this.root, key);
    }

    BSTNode insertRecur(BSTNode myRoot, int key){
        if (myRoot == null){
            myRoot = new BSTNode(key);
            return myRoot;
        }
        if (key < myRoot.key){
            myRoot.left = insertRecur(myRoot.left, key);
        }
        else if (key > myRoot.key){
            myRoot.right = insertRecur(myRoot.right, key);
        }
        return myRoot;
    }

    void displayInOrder(){
        displayRecur(this.root);
    }

    void displayRecur(BSTNode myRoot){
        if (myRoot != null){
            displayRecur(myRoot.left);
            System.out.println(myRoot.key);
            displayRecur(myRoot.right);
        }
    }

    public static void main(String[] args) {
        long startTime = System.currentTimeMillis();
        BST tree = new BST();
        Random random = new Random();

        for (int i = 0; i < 10; i++){
            int ranInt = random.nextInt(1000);

            tree.insert(ranInt);
        }

        tree.displayInOrder();
        long endTime = System.currentTimeMillis();

        System.out.println("Program took " + (endTime - startTime) + " ms...");
    }

}
class BSTNode{
    int key;
    BSTNode left;
    BSTNode right;

    public BSTNode(int number) {
        this.key = number;
        this.left = this.right = null;
    }
}